src/library/cratefeature.cpp

```diff
@@ -722,16 +769,19 @@ void CrateFeature::slotTrackSelected(TrackPointer pTrack) {
 
     // Set all crates the track is in bold (or if there is no track selected,
     // clear all the bolding).
-    int row = 0;
-    for (QList<QPair<int, QString> >::const_iterator it = m_crateList.begin();
-         it != m_crateList.end(); ++it, ++row) {
-        TreeItem* crate = rootItem->child(row);
-        if (crate == nullptr) {
-            continue;
+    QStack<TreeItem*> cratesToProcess;
+    TreeItem* crate;
+    cratesToProcess.push(rootItem);
+    while (! cratesToProcess.empty()) {
+        crate = cratesToProcess.pop();
+        
+        crate->setBold(
+                m_cratesSelectedTrackIsIn.contains(
+                    crate->dataPath().toInt()));
+
+        for(int row = 0; row < crate->childCount(); row++) {
+            cratesToProcess.push(crate->child(row));
```

Comment:
This might be likely a performance killer on low end hardware. 
Use a recursive lookup without a temporary stack